### ISSUE TITLE
Avoid scanning the complete set of entities in `FileSystemGraphObjectStore.getEntity({ _key: string })`

### ISSUE THREAD

**Issue by aiwilliams** [JupiterOne/sdk#408]
When requesting an entity having a known `_key` value, considering the effort that goes into preventing duplicate keys, and the fact that the filesystem isn't scanned at all when the entity has not been flushed to disk, answer the first found match and stop iteration.

I would consider this to be a bug as well, because in one case duplicate entries will be exposed, in another case it will not be exposed.

The current code:

```ts
  async getEntity({ _key, _type }: GraphObjectLookupKey): Promise<Entity> {
    const bufferedEntity = this.localGraphObjectStore.findEntity(_key);

    if (bufferedEntity) {
      // This entity has not yet been flushed to disk
      return bufferedEntity;
    }

    const entities: Entity[] = [];

    await this.iterateEntities({ _type }, async (e) => {
      if (e._key === _key) {
        entities.push(e);
      }

      return Promise.resolve();
    });

    if (entities.length === 0) {
      throw new IntegrationMissingKeyError(
        `Failed to find entity (_type=${_type}, _key=${_key})`,
      );
    } else if (entities.length > 1) {
      throw new IntegrationDuplicateKeyError(
        `Duplicate _key detected (_type=${_type}, _key=${_key})`,
      );
    } else {
      return entities[0];
    }
  }
```

- [ ] Stop iteration when a match by `_key` is found
- [ ] Update documentation for `findEntity` and `getEntity` to reflect that no `IntegrationDuplicateKeyError` will be thrown

**Comment by austinkelleher** [comment]
@aiwilliams We are on the same page :) https://github.com/JupiterOne/sdk/issues/385

**Comment by austinkelleher** [comment]
Fixed in #454
